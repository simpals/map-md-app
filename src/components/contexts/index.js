import { createContext } from 'react';

export const MapCtx = createContext({
  MapGL: null,
  userLocation: null,
  authorization: null,
  style: null,
  changeStyle: () => {}
});
