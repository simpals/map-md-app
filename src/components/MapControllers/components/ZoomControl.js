import React, { useContext } from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import { MapCtx } from '../../contexts';

const Wrapper = styled.div`
  position: absolute;
  right: 15px;
  top: 73px;
  margin: auto;
  width: 36px;
  height: 77px;
  border-radius: 19.5px;
  background-color: #ffffff;
  overflow: hidden;
  z-index: 100;
  box-shadow: 0 2px 4px 0 rgba(0, 0, 0, 0.1), 0 3px 10px 0 rgba(0, 0, 0, 0.07);
  ${({ position }) => (!position ? 'top: 16px' : '')}
  ${({ theme: { zoomControl } }) => zoomControl && zoomControl}
`;

const ZoomBtn = styled.button`
  display: block;
  text-align: center;
  margin: 0;
  padding: 0;
  width: 100%;
  height: 50%;
  &:active {
    background: #fcfcfc;
  }
`;

const Separator = styled.span`
  width: 18px;
  height: 1px;
  background-color: #ededed;
  display: block;
  margin: 0 auto;
`;

const ZoomIconPlus = styled.span`
  font-size: 18px;
  color: rgba(0, 0, 0, 0.54);
`;

const ZoomIconMinus = styled.span`
  font-size: 18px;
  color: rgba(0, 0, 0, 0.54);
`;

const ZoomControl = ({ position }) => {
  const { MapGL } = useContext(MapCtx);

  const zoomIn = () => {
    const nextZoom = MapGL.getZoom() + 1;
    MapGL.zoomTo(nextZoom, null, { type: 'some' });
  };

  const zoomOut = () => {
    const nextZoom = MapGL.getZoom() - 1;
    MapGL.zoomTo(nextZoom, null, { type: 'some' });
  };

  return (
    <Wrapper position={position} className='ZoomControl' >
      <ZoomBtn type='button' onClick={zoomIn}>
        <ZoomIconPlus className='icon-add' />
      </ZoomBtn>
      <Separator />
      <ZoomBtn type='button' onClick={zoomOut}>
        <ZoomIconMinus className='icon-remove' />
      </ZoomBtn>
    </Wrapper>
  );
};

ZoomControl.propTypes = {
  position: PropTypes.bool.isRequired
};

export default ZoomControl;
