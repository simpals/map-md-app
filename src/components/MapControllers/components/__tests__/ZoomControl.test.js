import 'jsdom-global/register';
import React from 'react';
import { shallow, mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import ZoomControl from '../ZoomControl';
import { MapCtx } from '../../../contexts';

const props = {
  position: false
};

const context = {
  MapGL: {
    getZoom: jest.fn(),
    zoomTo: jest.fn()
  }
};

describe('test ZoomControl component', () => {
  it('renders without crashing', () => {
    const component = shallow(<ZoomControl {...props} />);
    expect(toJson(component)).toMatchSnapshot();
  });

  it('test zoomIn click', () => {
    const component = mount(
      <MapCtx.Provider value={context}>
        <ZoomControl {...props} />
      </MapCtx.Provider>
    );
    component.find('button').first().simulate('click');
  });

  it('test zoomOut click', () => {
    const component = mount(
      <MapCtx.Provider value={context}>
        <ZoomControl {...props} />
      </MapCtx.Provider>
    );
    component.find('button').last().simulate('click');
  });
});
