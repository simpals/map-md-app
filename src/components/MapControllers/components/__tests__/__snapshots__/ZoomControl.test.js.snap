// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test ZoomControl component renders without crashing 1`] = `
<styled.div
  className="ZoomControl"
  position={false}
>
  <styled.button
    onClick={[Function]}
    type="button"
  >
    <styled.span
      className="icon-add"
    />
  </styled.button>
  <styled.span />
  <styled.button
    onClick={[Function]}
    type="button"
  >
    <styled.span
      className="icon-remove"
    />
  </styled.button>
</styled.div>
`;
